Networking/Networking/Requests/DotcomRequest.swift

```diff
@@ -59,4 +100,37 @@ struct DotcomRequest: Request {
             return WordPressApiValidator()
         }
     }
+
+    func asRESTRequest(with siteURL: String) -> RESTRequest? {
+        guard availableAsRESTRequest else {
+            return nil
+        }
+
+        guard wordpressApiVersion.isWPOrgEndpoint else {
+            return nil
+        }
+
+        // As the REST request is directly sent to the site URL, we remove site info from path
+        guard let pathWithoutSiteInfo = try? pathAfterRemovingSitesComponent() else {
+            return nil
+        }
+
+        return RESTRequest(siteURL: siteURL,
+                           wooApiVersion: .none,
+                           method: method,
+                           path: wordpressApiVersion.path + pathWithoutSiteInfo,
+                           parameters: parameters ?? [:])
+    }
+}
+
+private extension DotcomRequest {
+    /// Removes the site info from the `path`
+    ///
+    /// - Returns: Path without `sites/$siteID/`
+    ///
+    func pathAfterRemovingSitesComponent() throws -> String {
+        let regex = try NSRegularExpression(pattern: "([\\/]*sites\\/.[^\\/]*\\/)")
```

Comment:
❓ I'm a bit rusty with regex, does the first part of the regex indicate that it's expecting a `/` at the beginning? if so, I'm not sure if we want that since the paths are usually without the front slash as the documentation suggests for this function.

`test_the_converted_RESTRequest_path_does_not_have_sites_component` is also using a path with a front slash, we might want to update that or add another test case if we want to support both with and without a slash.

Reply:
This will work for both paths with/without a front slash. Added unit tests in 643f3a57012c79985baf8bfb368ce59f6f9810b9